extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -1138,19 +1160,47 @@ private JsonObject getVideoSecondaryInfoRenderer() throws ParsingException {
                                     "-1")));
                             itagItem.setIndexEnd(Integer.parseInt(indexRange.getString("end",
                                     "-1")));
-                            itagItem.fps = formatData.getInt("fps");
                             itagItem.setQuality(formatData.getString("quality"));
                             itagItem.setCodec(codec);
+                            if (streamType != StreamType.VIDEO_STREAM) {
+                                itagItem.setTargetDurationSec(formatData.getInt(
+                                        "targetDurationSec"));
+                            }
+                            if (itagType == ItagItem.ItagType.VIDEO
+                                    || itagType == ItagItem.ItagType.VIDEO_ONLY) {
+                                itagItem.setFps(formatData.getInt("fps"));
+                            }
+                            if (itagType == ItagItem.ItagType.AUDIO) {
+                                itagItem.setSampleRate(Integer.parseInt(formatData.getString(
+                                        "audioSampleRate")));
```

Comment:
No, because YouTube return this value as a string!

<img src="https://user-images.githubusercontent.com/74829229/158467678-9c5191f4-3209-4585-8c42-0337abaf93d1.png" width="300px" alt="Sample rate return YT player response">

Reply:
Okay YT...

Thank you for the hint.

You should note that somewhere inside the code 😉

Reply:
Done